From 64c6a0abd7d00ffa149c1ec90fab1f664a21cb7e Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Wed, 18 Jul 2018 17:09:30 +0300
Subject: [PATCH 4176/5242] MLK-18951: ARM: dts: imx6qdl-sabresd: Fix
 duplicate leds node

commit  a2840e7ec8e7e459b6e7cab9dfb35dfcba4a75d1 from
https://source.codeaurora.org/external/imx/linux-imx.git

The charger led definition ported from imx_4.9.y use the same gpio with
another definition introduced by upstream commit 702bfbef1c98 ("ARM:
dts: imx6qdl-sabresd.dtsi: Add red led")

Fix by explicitly overwriting the upstream definition with ours.

Fixes: 491c8dc6801e ("MLK-11703-6: ARM: imx6: Add charging led support on Sabresd board")
Fixes: 4aada993e5be ("MLK-11703-7: ARM: dts: imx6qdl-sabresd: add retain-state-suspended property in dts")

This commit could be squashed in the above on next rebase.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Acked-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi |   15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index ddbb087..703040e 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -43,16 +43,6 @@
 		stdout-path = &uart1;
 	};
 
-	leds {
-		compatible = "gpio-leds";
-
-		charger-led {
-			gpios = <&gpio1 2 0>;
-			linux,default-trigger = "max8903-charger-charging";
-			retain-state-suspended;
-		};
-	};
-
 	memory: memory@10000000 {
 		reg = <0x10000000 0x40000000>;
 	};
@@ -259,9 +249,10 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_gpio_leds>;
 
-		red {
+		charger-led {
 			gpios = <&gpio1 2 0>;
-			default-state = "on";
+			linux,default-trigger = "max8903-charger-charging";
+			retain-state-suspended;
 		};
 	};
 
-- 
1.7.9.5

